### PR TITLE
feature/slow-stochastic-indicator

### PR DESCRIPTION
**Pull request by pawelfus**
Added Slow Stochastic indicator.

### CODE REVIEW

File: ts/indicators/slow-stochastic.src.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+                    this,
+                    ({
+                        xData: slowValues.xData,
+                        yData: fastYData
+                    } as any),
+                    {
+                        index: 1,
+                        period: periods[2]
+                    }
+                );
```
Comment (pawelfus) [inline ts/indicators/slow-stochastic.src.ts]: Can I do something with this little monster? Especially lines 137-139.. @bre1470

File: ts/indicators/slow-stochastic.src.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            this: Highcharts.SlowStochasticIndicator,
+            series: TLinkedSeries,
+            params: Highcharts.SlowStochasticIndicatorParamsOptions
+        ): (Highcharts.IndicatorValuesObject<TLinkedSeries>|undefined) {
+            const periods: Array<number> = (params.periods as any),
+                fastValues = seriesTypes.stochastic.prototype.getValues.call(
+                    this,
+                    series,
+                    params
+                ) as any,
```
Comment (pawelfus) [inline ts/indicators/slow-stochastic.src.ts]: Decided to cast to any, I couldn't resolve that issue. It seems that getting values from other indicators is a bit cumbersome. FYI @bre1470

File: ts/indicators/slow-stochastic.src.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        class SlowStochasticIndicator extends StochasticIndicator {
+            public nameBase: string;
+            public init(): void;
```
Comment (pawelfus) [inline ts/indicators/slow-stochastic.src.ts]: I assume that I should define here all methods that are overwritten, right? @bre1470

Comment (bre1470) [reply ts/indicators/slow-stochastic.src.ts]: Yes, except if you do a super call with all arguments (`apply(this, arguments)`).

File: ts/indicators/slow-stochastic.src.ts (hunk 1 above)
Comment (bre1470) [reply ts/indicators/slow-stochastic.src.ts]: Please see `ts/README.md` and make use of parentheses with type unions. We have to discuss the increase of line characters though.

File: ts/indicators/slow-stochastic.src.ts (hunk 2 above)
Comment (bre1470) [reply ts/indicators/slow-stochastic.src.ts]: If I understand it correctly, stochastic is expecting a two-dimensional series (OHLC), but limiting the generic variable `TValueSeries` leads to inheritence conflict. So I agree with that.

File: ts/indicators/slow-stochastic.src.ts (hunk 1 above)
Comment (pawelfus) [reply ts/indicators/slow-stochastic.src.ts]: Ahh, I see, will do, thanks!

File: ts/indicators/slow-stochastic.src.ts (hunk 2 above)
Comment (pawelfus) [reply ts/indicators/slow-stochastic.src.ts]: Maybe somewhere `Highcharts.IndicatorValuesObject<TLinkedSeries>` makes this incompatible?

Just tried to compare with `ppo` indicator, where it's first cast as `IndicatorValuesObject`, then casted to `any`:
```js
            LPE = EMA.prototype.getValues.call(this, series, {
                index: index,
                period: periods[1]
            }) as Highcharts.IndicatorValuesObject<TLinkedSeries>;

            ...

            for (...) {
                PPO.push([(LPE as any).xData[i], oscillator]);
                xData.push((LPE as any).xData[i]);
                yData.push(...);
            }
```

Anyway, if this makes sense to you @bre1470 then I'm fine :)

File: ts/indicators/slow-stochastic.src.ts (hunk 3 above)
Comment (pawelfus) [reply ts/indicators/slow-stochastic.src.ts]: Ok, thanks! 👍

**Pull request merged**